File: lv1-library-loan/csharp/LibraryLoan.Domain/Members/LoanRecordErrors.cs
using LibraryLoan.Results;

namespace LibraryLoan.Domain.Members;

public static class LoanRecordErrors
{
    public static Error NotFound(LoanRecordId loanRecordId) => new(
        "LoanRecord.NotFound",
        $"貸出記録(ID: {loanRecordId})が見つかりません。");

    public static Error AlreadyReturned(LoanRecordId loanRecordId) => new(
        "LoanRecord.AlreadyReturned",
        $"貸出記録(ID: {loanRecordId})は既に返却済みです。");
}

File: lv1-library-loan/typescript/src/domain/members/loan-record-errors.ts
import { ResultError } from "../../results/result-error.js";
import type { LoanRecordId } from "./loan-record-id.js";

export const LoanRecordErrors = {
  notFound: (loanRecordId: LoanRecordId): ResultError =>
    new ResultError("LoanRecord.NotFound", `貸出記録(ID: ${loanRecordId})が見つかりません。`),

  alreadyReturned: (loanRecordId: LoanRecordId): ResultError =>
    new ResultError("LoanRecord.AlreadyReturned", `貸出記録(ID: ${loanRecordId})は既に返却済みです。`),
};
